app/main.py

```diff
@@ -0,0 +1,63 @@
+import json
+from typing import MutableMapping
+from uuid import UUID
+
+import uvicorn
+import yaml
+from fastapi import FastAPI, Query, HTTPException
+from sdc.crypto.jwe_helper import JWEHelper
+from sdc.crypto.key_store import KeyStore
+
+KEY_PURPOSE_SDS = "supplementary_data"
+
+app = FastAPI()
+
+with open("dev-keys.yml", encoding="UTF-8") as keys_file:
+    keys = KeyStore(yaml.safe_load(keys_file))
+
+
+@app.get("/v1/unit_data")
+def get_sds_data(
+    dataset_id: UUID = Query(), identifier: str = Query(min_length=1)
+) -> MutableMapping:
+    # The mock current does not make use of identifier
+    guid_filename_map = {
+        "c067f6de-6d64-42b1-8b02-431a3486c178": "supplementary_data_no_repeat",
+        "693dc252-2e90-4412-bd9c-c4d953e36fcd": "supplementary_data_no_repeat_v2",
+        "34a80231-c49a-44d0-91a6-8fe1fb190e64": "supplementary_data_with_repeat",
+        "6b378962-f0c7-4e8c-947e-7d24ee1b6b88": "supplementary_data_with_repeat_v2",
+    }
+
+    if filename := guid_filename_map.get(str(dataset_id)):
+        return encrypt_mock_data(load_mock_data(f"mock_data/{filename}.json"))
+
+    raise HTTPException(status_code=404)
+
+
+@app.get("/v1/dataset_metadata")
+def get_sds_dataset_ids(survey_id: str = Query()) -> list[dict]:
```

Comment:
I think this one needs `period_id` as well

Reply:
> I think this one needs period_id as well (as per you noticing the missing value in the main one)

Done.

> just for understanding, why add = Query()? just for consistency with the other parameters?

Kept this but removed the dataset_id one as I am using UUID type now. Rest of the query one have validation including this one now.

Reply:
Yep makes sense for the validation ones 👍 nice

Reply:
Will resolve the comment when I have permissions to do so 😄 or feel free